### ISSUE TITLE
should we add a 'keep running if you have a failure mode' option for `helm install`?

### ISSUE THREAD

**Issue by jstrachan** [helm/helm#318]
Right now `helm install` terminates if any `kubectl create` fails.

As we start to make composite apps with different RCs and Services that can overlap. New versions of a chart may add a new Service or RC for example.

Users might want a way to try to create all the kubernetes resources inside a chart; then summarise all the failures at the end; rather than failing fast. e.g. so that any resources that don't exist will be created. e.g. if you did 

```
helm install cheese
# time passes a new RC called 'bar' is added to new version of cheese
helm update
helm install cheese
```
The second install fails with duplicates for all the things in the first version of cheese; but the 2nd install will at least install 'bar'

Another option could be to try check if each resource exists already before doing the `kubectl create` so that we can avoid trying to create something that already exists?

**Comment by arschles** [comment]
I like the last option you mentioned. having `helm install` fail fast, and providing a `helm upgrade` (as discussed in #292) makes the most sense to me, and follows what homebrew and yum (iirc, it's been a while) both do.

**Comment by technosophos** [comment]
Good idea. We already have the `force` flag. Should this behavior be rolled into that, or should we create a separate flag?

**Comment by jstrachan** [comment]
@arschles if you want to play, there's an implementation of `helm upgrade` that works as you describe in this PR: https://github.com/helm/helm/pull/323
